### PR TITLE
Convert HALClustering to Fast

### PR DESCRIPTION
**Pull request by AlexeyVatolin**
To solve issue #814

## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Review by isaac-chung — APPROVED** [review]
Thanks for adding this. Please add review points and it's good to merge!

**Comment by AlexeyVatolin** [comment]
I'll write it down for the story. To get information about the date range in the dataset, I parsed again all articles from https://hal.science/<hal_id> and took there the date of publication of the article. Hal_id is a column from dataset

File: mteb/tasks/Clustering/fra/HALClusteringS2S.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
+from mteb.abstasks.AbsTaskClustering import AbsTaskClustering
+from mteb.abstasks.AbsTaskClusteringFast import AbsTaskClusteringFast
 from mteb.abstasks.TaskMetadata import TaskMetadata
 
-from ....abstasks.AbsTaskClustering import AbsTaskClustering
+NUM_SAMPLES = 2048
 
 
 class HALClusteringS2S(AbsTaskClustering):
+    superseeded_by = "HALClusteringS2SFast"
```
Comment (isaac-chung) [inline mteb/tasks/Clustering/fra/HALClusteringS2S.py:14]: Nice catch.
Sorry that i missed this the first time. We want to be consistent with the task name so this should be 
```suggestion
    superseeded_by = "HALClusteringS2S.v2"
```

The class name can remain the same.

**Review by isaac-chung — COMMENTED** [review]
Thanks again. Just a small comment I missed last time.

File: mteb/tasks/Clustering/fra/HALClusteringS2S.py [hunk 2]
```diff
@@ -46,3 +50,64 @@ def dataset_transform(self):
...
+        eval_splits=["test"],
+        eval_langs=["fra-Latn"],
+        main_score="v_measure",
+        date=("2000-03-29", "2024-05-24"),
+        form=["written"],
+        domains=["Academic"],
+        task_subtypes=["Thematic clustering"],
+        license="Apache-2.0",
+        socioeconomic_status="medium",
+        annotations_creators="derived",
```
Comment (imenelydiaker) [inline mteb/tasks/Clustering/fra/HALClusteringS2S.py:75]: Thank you for filling this!
HAL is like Arxiv for French academics, I think annotations are created by authors of the articles when submitting, can we say it's `human-annotated` then?

File: mteb/tasks/Clustering/fra/HALClusteringS2S.py [hunk 3]
```diff
@@ -46,3 +50,64 @@ def dataset_transform(self):
...
+
+    def dataset_transform(self):
+        """Convert to standard format"""
+        self.dataset["test"] = self.dataset["test"].remove_columns("hal_id")
+        self.dataset["test"] = self.dataset["test"].rename_columns(
+            {"title": "sentences", "domain": "labels"}
+        )
+        labels_count = Counter(self.dataset["test"]["labels"])
+
+        # leave classes with more than 2 samples after stratified_subsampling
```
Comment (imenelydiaker) [inline mteb/tasks/Clustering/fra/HALClusteringS2S.py:91]: You mean keep classes with more than 2 samples?

File: mteb/tasks/Clustering/fra/HALClusteringS2S.py (hunk 1 above)
Comment (AlexeyVatolin) [reply mteb/tasks/Clustering/fra/HALClusteringS2S.py:14]: Ok, I'll change it here, but I already merged [pull request](https://github.com/embeddings-benchmark/mteb/pull/813/files) with name Fast

File: mteb/tasks/Clustering/fra/HALClusteringS2S.py (hunk 3 above)
Comment (AlexeyVatolin) [reply mteb/tasks/Clustering/fra/HALClusteringS2S.py]: Yes

File: results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/HALClusteringS2S.v2.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            0.20506254032611254,
+            0.22693097512239063,
+            0.2586225722627258,
+            0.2794849957576204
+          ]
+        }
+      }
+    ]
+  },
+  "task_name": "HALClusteringS2SFast"
```
Comment (isaac-chung) [inline results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/HALClusteringS2S.v2.json:32]:
```suggestion
  "task_name": "HALClusteringS2S.v2"
```

File: results/intfloat__multilingual-e5-small/HALClusteringS2S.v2.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            0.17714878319922156,
+            0.17265039444664382,
+            0.20571471031898306,
+            0.19523564981493394
+          ]
+        }
+      }
+    ]
+  },
+  "task_name": "HALClusteringS2SFast"
```
Comment (isaac-chung) [inline results/intfloat__multilingual-e5-small/HALClusteringS2S.v2.json:32]:
```suggestion
  "task_name": "HALClusteringS2S.v2"
```

**Review by isaac-chung — COMMENTED** [review]
task names should follow the new name as well.

**Pull request merged**